src/umockdev.vala

```diff
@@ -687,6 +714,50 @@ public class Testbed: GLib.Object {
         remove_dir(real_path, true);
     }
 
+    private string? find_a_child(string syspath)
```

Comment:
I find this too complicated, and repeating too much logic. It feels to me that this should be a single `remove_device()` that just recursively calls itself (depth-first!). If you need extra arguments, make it a remove_device_with_children() and then remove_device() becomes an one-liner which calls the former.

Reply:
Sorry, I will re-work.